dspam_ fixes

jolan78 opened the pull request:
- typo
- cleaning invalid lockfile
- new graph relprocessed in messages / minute

sumpfralle commented on plugins/dspam/dspam_:
```diff
@@ -211,7 +215,15 @@ file_is_locked() {
 	local lock=$1.lock
 
-	[ -f $lock ] && debug file $file is locked && return 0 # EX_OK
+	if [ -f $lock ];then
```
Please quote variables, whenever it is possible.
Probably `shellcheck -s dash dspam_` would help you finding all occurrences.

sumpfralle commented on plugins/dspam/dspam_:
```diff
@@ -211,7 +215,15 @@ file_is_locked() {
-	[ -f $lock ] && debug file $file is locked && return 0 # EX_OK
+	if [ -f $lock ];then
+		local pid=$(cat $1.lock)
+		if [ $(ps h -p $pid -o comm=|grep -c dspam_) -eq 0 ]; then
```
Maybe the following is easier to read?
```if ps h -p "$pid" -o comm= | grep -q dspam_; then```
(and switch both branches)

sumpfralle commented on plugins/dspam/dspam_:
```diff
@@ -211,7 +215,15 @@ file_is_locked() {
+			debug file $file is locked && return 0 # EX_OK
+		fi
+	fi
 	debug file $file is not locked
```
The following two lines should probably move to the "no lock" branch above?

sumpfralle commented on plugins/dspam/dspam_:
```diff
@@ -211,7 +215,15 @@ file_is_locked() {
+			debug lock for file $file is no longer valid
+			file_remove_lock $file
+		else
+			debug file $file is locked && return 0 # EX_OK
```
Maybe split both commands?
This will make the flow-breaking `return` a bit more obvious.
Additionally it removes the conditional relationship between these (unrelated) commands.

sumpfralle commented on plugins/dspam/dspam_:
```diff
@@ -452,26 +465,34 @@ print_config() {
 			echo graph_category spamfilter
 			echo graph_args --base 1000
-			echo graph_vlabel Messages
+			[ $graph == absprocessed ] && echo graph_vlabel Messages
```
This plugin uses `/bin/sh` (not `/bin/bash`) - thus you should avoid bashisms:
```[ "$graph" = absprocessed ] && echo graph_vlabel Messages```

jolan78 commented:
I updated the commits according to your observations.
regarding quoted variables, I was trying to stick to the general code style of the file. I fixed some of the unquoted variables but didn't change those which are unquoted in the remaining code. tell me if that's OK for you.

sumpfralle commented:
It is much better thank before - thank you!

sumpfralle commented:
btw: I am heavily confused by the usage of the "statsfile" within in this plugin (a third of the plugin code deals with the locking of that file). As far as I understand the code, the statsfile is used only for storing intermediate processing results. Thus this data could be easily stored in a simple variable instead of working around temporary files and locks. Or do I miss something?
(just being curious)

jolan78 commented:
the statefile is used to avoid running dspam_stats too much. if the file is less than 3 minutes old (by default) the file is not generated again. it's usefull when you have several graphs type , when you want one graph per user , or when you have several munin instances pulling the data from your server.

sumpfralle commented:
Ah - I see. Thank you!
